src/middlewared/middlewared/plugins/vm/vm_devices.py

```diff
@@ -48,6 +49,21 @@ class Config:
         datastore_extend = 'vm.device.extend_device'
         cli_namespace = 'service.vm.device'
 
+    @accepts()
+    @returns(Dict(additional_attrs=True, example={'vms/test 1': '/dev/zvol/vms/test+1'}))
+    async def disk_choices(self):
+        """
+        Returns disk choices for device type "DISK".
+        """
+        return {
+            vol['id']: os.path.join('/dev/zvol', vol['id'].replace(' ', '+'))
```

Comment:
Shouldn't keys be the encoded value (with `+`) and values be human-readable value? Also let's use `zvol_name_to_path` here.